lib/web3/requestmanager.js

```diff
@@ -226,9 +226,16 @@ RequestManager.prototype.poll = function () {
         }
 
         results.map(function (result, index) {
-            result.callback = self.polls[index].callback;
-            return result;
+            // make sure the filter is still installed after arrival of the request
+            if(self.polls[index]) {
```

Comment:
this is not a solution. `self.polls` is an Array and you may still call callback for different filter.